### PR TITLE
MODINVOICE-53- Adjustment schema changes

### PR DESCRIPTION
**Pull request by KVupp**
## Purpose
https://issues.folio.org/browse/MODINVOICE-53 - Adjustment schema changes
## Approach
1. Rename fields
2. Modify examples

#### TODOS and Open Questions
- [x] Create corresponding changes in mod-invoice after [#20](https://github.com/folio-org/mod-invoice/pull/20) is merged

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Examples exist for all schemas
  - [x] Descriptions exist for all schema properties
  - [x] All schemas pass raml-cop linting
- Does this introduce breaking changes?
  - [ ] Were there any schema changes?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: mod-invoice-storage/schemas/adjustment.json [hunk 1]
```diff
@@ -3,26 +3,65 @@
...
+      ],
+      "default" : "Not prorated"
+    },
+    "relationToTotal": {
+      "description": "Relationship of this adjustment to the total;In addition to: added to subtotal;Included in: reported as subtotal portion;Separate from:calculated from subtotal",
+      "type": "string",
+      "enum": [
+        "In addition to",
+        "Included in",
+        "Separate from"
```
Comment (craigmcnally) [inline mod-invoice-storage/schemas/adjustment.json]: Don't we also need "Not prorated" in the enum list too?

Comment (craigmcnally) [reply mod-invoice-storage/schemas/adjustment.json]: oh wait, I'm looking at the wrong field.. but I don't think we want "Not prorated" here as the default... i think we want "In addition to"

File: mod-invoice-storage/examples/invoice_line_collection.sample [hunk 2]
```diff
@@ -4,14 +4,18 @@
       "id": "e0d08448-343b-118a-8c2f-4fb50248d672",
```
Comment (craigmcnally) [inline mod-invoice-storage/examples/invoice_line_collection.sample]: Just left a comment on Piotr's PR about reusing UUIDs... So to be fair I'll leave one here too ;)

File: mod-invoice-storage/examples/invoice.sample [hunk 3]
```diff
@@ -2,17 +2,22 @@
   "id": "733cafd3-895f-4e33-87b7-bf40dc3c8069",
   "adjustments": [
     {
-      "adjustmentLabel": "Sales Tax",
-      "adjustmentType": "Percentage",
-      "adjustmentValue": 6
+       "id":"14263aab-b22b-4ddc-9ecc-3434427c2c8f"
```
Comment (craigmcnally) [inline mod-invoice-storage/examples/invoice.sample]: invoice adjustments which aren't prorated and are "in addition to" the total... this is where you'd want (need?) a fundDistribution.  We need to know which fund to pay this adjustment from.

File: mod-invoice-storage/examples/invoice.sample [hunk 4]
```diff
@@ -2,17 +2,22 @@
...
+       "type":"Amount",
+       "value":4.50,
+       "prorate":"By line",
+       "relationToTotal":"In addition to"
     },
     {
-      "adjustmentLabel": "Shipping",
-      "adjustmentType": "Amount",
-      "adjustmentValue": 8.50
+       "description":"Some Tax",
```
Comment (craigmcnally) [inline mod-invoice-storage/examples/invoice.sample]: invoice adjustments which aren't prorated and are "in addition to" the total... this is where you'd want (need?) a fundDistribution. We need to know which fund to pay this adjustment from.

**Comment by craigmcnally** [comment]
Note https://github.com/folio-org/mod-invoice/pull/20 will be affected by these changes

File: mod-invoice-storage/examples/invoice.sample (hunk 4 above)
Comment (KVupp) [reply mod-invoice-storage/examples/invoice.sample]: I can't wait for the fund distribution calculation stories  :P

**Comment by KVupp** [comment]
> Note folio-org/mod-invoice#20 will be affected by these changes
I added a TODO above to create a PR after those changes are merged

File: mod-invoice-storage/examples/invoice.sample [hunk 5]
```diff
@@ -2,17 +2,29 @@
   "id": "733cafd3-895f-4e33-87b7-bf40dc3c8069",
   "adjustments": [
     {
-      "adjustmentLabel": "Sales Tax",
-      "adjustmentType": "Percentage",
-      "adjustmentValue": 6
+       "id":"14263aab-b22b-4ddc-9ecc-3434427c2c8f"
```
Comment (piotr-kalashuk) [inline mod-invoice-storage/examples/invoice.sample]: Comma is missing in the end of line

File: mod-invoice-storage/examples/invoice_collection.sample [hunk 6]
```diff
@@ -4,9 +4,12 @@
       "id": "c0d08448-347b-418a-8c2f-5fb50248d67e",
       "adjustments": [
         {
-          "adjustmentLabel": "Shipping",
-          "adjustmentType": "Amount",
-          "adjustmentValue": 4.50
+          "id":"14263aab-b22b-4ddc-9ecc-3434427c2c8f"
```
Comment (piotr-kalashuk) [inline mod-invoice-storage/examples/invoice_collection.sample]: Comma is missing in the end of line

File: mod-invoice-storage/examples/invoice_line.sample [hunk 7]
```diff
@@ -1,23 +1,31 @@
...
+      "type":"Percentage",
+      "value":8,
+      "prorate":"Not prorated",
+      "relationToTotal":"In addition to"
     },
     {
-      "adjustmentLabel": "Shipping",
-      "adjustmentType": "Amount",
-      "adjustmentValue": 2.50
+      "id":"14263aab-b22b-4ddc-9ecc-3434427c2c8f"
```
Comment (piotr-kalashuk) [inline mod-invoice-storage/examples/invoice_line.sample]: Comma is missing in the end of line

File: mod-invoice-storage/examples/invoice_line_collection.sample [hunk 8]
```diff
@@ -4,14 +4,18 @@
...
       "adjustments": [
         {
-          "adjustmentLabel": "Service Fee",
-          "adjustmentType": "Amount",
-          "adjustmentValue": 2.25
+          "description":"Service Fee",
+          "type":"Amount",
+          "value":2.25,
+          "prorate":"Not Prorated",
```
Comment (piotr-kalashuk) [inline mod-invoice-storage/examples/invoice_line_collection.sample]: `Not Prorated` should be changed to `Not prorated`

File: mod-invoice-storage/examples/invoice_line_collection.sample [hunk 9]
```diff
@@ -4,14 +4,18 @@
...
+          "relationToTotal":"In addition to"
         },
         {
-          "adjustmentLabel": "Shipping",
-          "adjustmentType": "Amount",
-          "adjustmentValue": 2.75
+          "description":"Shipping",
+          "type":"Amount",
+          "value":2.75,
+          "prorate":"Not Prorated",
```
Comment (piotr-kalashuk) [inline mod-invoice-storage/examples/invoice_line_collection.sample]: `Not Prorated` should be changed to `Not prorated`

File: mod-invoice-storage/schemas/adjustment.json [hunk 10]
```diff
@@ -3,26 +3,65 @@
   "description": "Adjustment",
   "type": "object",
   "properties": {
-    "adjustmentLabel": {
+    "id": {
```
Comment (piotr-kalashuk) [inline mod-invoice-storage/schemas/adjustment.json]: According to [MODINVOICE-54](https://issues.folio.org/browse/MODINVOICE-54):
> *NOTE* The invoiceLine adjustments maintain a link back to the invoice adjustment. This is needed in order to allow for the invoice adjustment to be modified/removed and have those changes trickle down into the invoiceLine adjustments.

Looking at the samples there I see `id` in the invoice level adjustment:
```json
{
  "id":"14263aab-b22b-4ddc-9ecc-3434427c2c8f",
  "adjustmentLabel":"Service Fee",
  "adjustmentType":"Amount",
  "adjustmentValue":10,
  "prorate":"By line"
}
```
`adjustmentId` in the invoice line level adjustment
```json
{
  "adjustmentId":"14263aab-b22b-4ddc-9ecc-3434427c2c8f",
  "adjustmentLabel":"Service Fee",
  "adjustmentType":"Amount",
  "adjustmentValue":5,
  "prorate":"By line"
}
```
@craigmcnally is this schema change going to be made in separate story?

Comment (craigmcnally) [reply mod-invoice-storage/schemas/adjustment.json]: I missed this. We might as well make the change now.

Comment (KVupp) [reply mod-invoice-storage/schemas/adjustment.json]: Why can't they have the same field name "id" , it can still refer to the invoice level adjustment

Comment (craigmcnally) [reply mod-invoice-storage/schemas/adjustment.json]: I think it's confusing to have two adjustments (in invoice and invoice-line) with the same value in "id".  Having a separate "adjustmentsId" makes it clearer (at least to me) that we're referencing another adjustment record.  "id" is typically used to say this is the identifier for *this* record, not some other record.

**Comment by KVupp** [comment]
The changes were left out in my local environment. Pushed the adjustmentId changes

**Pull request merged**